test/test_sparse_csr.py

```diff
@@ -551,12 +551,22 @@ def test_empty_like(self, layout, layout2, device, dtype):
     @all_sparse_compressed_layouts()
     @dtypes(*all_types_and_complex_and(torch.half, torch.bool, torch.bfloat16))
     def test_validate(self, layout, device, dtype):
+        def make_zero_batched(t):
+            return torch.empty(*((0,) + t.shape), dtype=t.dtype, device=t.device)
+
         for index_dtype in [torch.int32, torch.int64]:
             for (compressed_indices, plain_indices, values), kwargs in self.generate_simple_inputs(
                     layout, device=device, dtype=dtype, index_dtype=index_dtype, output_tensor=False):
                 size = kwargs['size']
                 torch._validate_sparse_compressed_tensor_args(compressed_indices, plain_indices, values, size, layout)
 
+                # check empty batch
+                torch._validate_sparse_compressed_tensor_args(
+                    *map(lambda t: make_zero_batched(t), (compressed_indices, plain_indices, values)),
+                    (0,) + size,
+                    layout
+                )
+
```

Comment:
Instead of the changes here, the `generate_simple_inputs` method ought to be updated to include zero-sized batch samples:
https://github.com/pytorch/pytorch/blob/cbfed470bd9ef845a9670e21d67f60cb44828062/torch/testing/_internal/common_utils.py#L2855

Reply:
Thank you, @pearu, I will do it as a follow-up. We also cannot create empty-batched compressed formats from empty dense inputs:
```python
>>> import torch
>>> torch.rand(0, 4, 4).to_sparse_bsr(2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: to_sparse_bsr: Expected product of batch dimensions to be non-zero.

```
and `zeros_like` is also broken for such inputs as per https://github.com/pytorch/pytorch/issues/101078.